extensions/interactions/CodeRepl/CodeRepl.js

```diff
@@ -114,27 +119,37 @@ oppia.directive('oppiaInteractiveCodeRepl', [
           });
         };
 
-        var fixLineNumbers = function(err) {
-          var preCodeNumLines = $scope.preCode.split('\n').length;
-          var userCodeNumLines = $scope.code.split('\n').length;
-          return err.replace(/on line ([0-9]+)/g, function(match, p1) {
-            var originalLineNumber = Number(p1);
-
-            return (
-              originalLineNumber <= preCodeNumLines ?
-              '(on line ' + String(originalLineNumber) +
-                ' of exploration creator\'s prepended code)' :
-              originalLineNumber <= preCodeNumLines + userCodeNumLines ?
-              '(on line ' + String(originalLineNumber - preCodeNumLines) + ')' :
-              '(on line ' + String(
-                  originalLineNumber - preCodeNumLines - userCodeNumLines) +
-                ' of exploration creator\'s appended code)');
-          });
-        };
+        var initMarkers =function(editor) {
+            var doc = editor.getDoc();
+            // The -1's here are because pre- and postCode end and start with a newline respectively.
+            var preCodeNumLines = $scope.preCode.split('\n').length - 1;
+            var postCodeNumLines = $scope.postCode.split('\n').length;
+            var fullCodeNumLines = $scope.code.split('\n').length;
+            var userCodeNumLines = (
+                fullCodeNumLines - preCodeNumLines - postCodeNumLines);
+            //mark pre- and post- code as uneditable, and give it some styling.
+            var markOptions = {
+                atomic: true,
+                readOnly: true,
+                className: "code-repl-marker",
+                inclusiveLeft: true,
+                inclusiveRight: true
+            }
+
+            doc.markText({line:0, ch:0},
+                         {line: preCodeNumLines, ch:0},
+                         Object.assign({}, markOptions, {inclusiveRight: false})
```

Comment:
In Chrome 43.0 I am getting this error:

   Object.assign is not a function

Can we use a different construct?